frontend/app/enrollment/layout.tsx

```diff
@@ -1,23 +1,36 @@
 "use client"
 import React from 'react'
 import useEnrollmentStore from '@/lib/store/enrollment-store'
-import { ChevronRight } from 'lucide-react'
-import { usePathname } from 'next/navigation'
+import {ChevronRight} from 'lucide-react'
+import useTaskProgressStore from "@/lib/store/task-progress-store";
 
 // TODO: Change the server component when we use access_tokens (no longer rely on the session) to fetch data
-export default function EnrollmentLayout({ children }: { children: React.ReactNode }) {
-    const { serviceRequest } = useEnrollmentStore()
-    const pathname = usePathname()
+export default function EnrollmentLayout({children}: { children: React.ReactNode }) {
+    const {serviceRequest} = useEnrollmentStore()
+    const {task} = useTaskProgressStore()
 
-    const isFirstStep = pathname === '/enrollment/new'
-    const isLastStep = pathname === '/enrollment/success'
+    const lastStepTaskStates = ['accepted', 'in-progress', 'rejected', 'failed', 'completed', 'cancelled', 'on-hold']
+    const isFirstStep = task?.status == "requested"
+    const isLastStep = task ? lastStepTaskStates.includes(task.status) : false
     const service = serviceRequest?.code?.coding?.[0].display
+    const statusTitles: Map<String, String> = new Map([
+        ["ready", service ? `${service} instellen` : "Instellen"],
+        ["requested", service ? `${service} instellen` : "Instellen"],
+        ["received", service ? `${service} instellen` : "Instellen"],
+        ["accepted", "Verzoek geaccepteerd"],
+        ["in-progress", "Verzoek in behandeling"],
+        ["on-hold", "Uitvoering gepauzeerd"],
+        ["completed", "Uitvoering afgerond"],
+        ["cancelled", "Uitvoering geannuleerd"],
+        ["failed", "Uitvoering mislukt"],
+        ["rejected", "Verzoek afgewezen"],
+    ])
```

Comment:
Ideally this is "just" a JSON object. This way we can also map the keys onto the task.status and make use of the type-safety to ensure we cover all statussen.

```suggestion
    const statusTitles = {
        "ready": service ? `${service} instellen` : "Instellen",
        "requested": service ? `${service} instellen` : "Instellen",
        "received": service ? `${service} instellen` : "Instellen",
        "accepted": "Verzoek geaccepteerd",
        "in-progress": "Verzoek in behandeling",
        "on-hold": "Uitvoering gepauzeerd",
        "completed": "Uitvoering afgerond",
        "cancelled": "Uitvoering geannuleerd",
        "failed": "Uitvoering mislukt",
        "rejected": "Verzoek afgewezen",
        "draft": "Verzoek in concept",
        "entered-in-error": "Verzoek gemarkeerd als foutief",
    }
```

Reply:
This failed earlier, but now it does compile... Probably I screwed up something else earlier. Changed it.